backend/jobs/log-alerts/log-alerts.go

```diff
@@ -0,0 +1,194 @@
+package log_alerts
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/highlight-run/highlight/backend/alerts"
+	"github.com/highlight-run/highlight/backend/clickhouse"
+	modelInputs "github.com/highlight-run/highlight/backend/private-graph/graph/model"
+	"github.com/highlight-run/highlight/backend/redis"
+	"github.com/highlight-run/highlight/backend/timeseries"
+	"github.com/openlyinc/pointy"
+
+	"github.com/pkg/errors"
+	"github.com/sendgrid/sendgrid-go"
+
+	"github.com/highlight-run/go-resthooks"
+	Email "github.com/highlight-run/highlight/backend/email"
+	"github.com/highlight-run/highlight/backend/model"
+	"github.com/highlight-run/highlight/backend/zapier"
+	log "github.com/sirupsen/logrus"
+	"gorm.io/gorm"
+)
+
+var AlertFrequencies = []int{15, 60, 300, 900, 1800}
+
+func WatchLogAlerts(ctx context.Context, DB *gorm.DB, TDB timeseries.DB, MailClient *sendgrid.Client, rh *resthooks.Resthook, redis *redis.Client, ccClient *clickhouse.Client) {
+	log.WithContext(ctx).Info("Starting to watch Log Alerts")
+
+	alertsByFrequency := &map[int][]*model.LogAlert{}
+	go func() {
+		// Every minute, check for new alerts and bucket by frequency
+		for range time.Tick(time.Minute) {
+			bucketed := map[int][]*model.LogAlert{}
+			for _, freq := range AlertFrequencies {
+				bucketed[freq] = []*model.LogAlert{}
+			}
+
+			alerts := getLogAlerts(ctx, DB)
+			for _, alert := range alerts {
+				for _, freq := range AlertFrequencies {
+					if freq >= alert.Frequency {
+						bucketed[freq] = append(bucketed[freq], alert)
+					}
+				}
+			}
+
+			alertsByFrequency = &bucketed
+		}
+	}()
+
+	for _, freq := range AlertFrequencies {
+		f := freq
+		go func() {
+			for range time.Tick(time.Duration(f) * time.Second) {
+				alerts := (*alertsByFrequency)[f]
+				if len(alerts) > 0 {
+					go func() {
```

Comment:
nit - don't need the extra func closure, can just call the method

Reply:
good point! my intention was to run each alert concurrently but this is operating on the list of alerts, will refactor